### ISSUE TITLE
Android - Video player got stuck loading and never started playing the video

### ISSUE THREAD

**Issue by Chamersoun** [muxinc/videojs-mux-kit#39]
Here is an code example I use for the testing - https://codesandbox.io/s/dark-moon-pb7qn?file=/src/App.js
The code is taken from the repo here - /src/demo/basic.html

The player is working just fine on different devices/browsers except the Chrome browser on iOS devices. The video got stuck loading. There is also an error in the sandbox - "Unknown browser" if you try to expand the video-player.

Failed:
iPhone XR - Chrome 92.04515.90
iPhone 7 - Chrome 92.04515.90

Passed:
iPhone XR - Firefox Daylight 36.0 (5732)
iPhone XR - Safari 14.7.1
iPhone 7 - Safari 14.7.1
Redmi Note 10S - Firefox Daylight 91.2.0
Redmi Note 10S - Chrome 92.0.4515.159
Redmi Note 10S - Edge 46.06.4.5161

**Comment by dylanjha** [comment]
@Chamersoun thanks for reporting this is [fixed in 0.4.0](https://github.com/muxinc/videojs-mux-kit/releases/tag/v0.4.0)!

**Comment by Chamersoun** [comment]
@dylanjha Awesome, thank you!

**Comment by dylanjha** [comment]
Hi @Chamersoun thanks for re-surfacing, we'll get the right fix in place. On any of these devices are you able to try opening this URL directly in the browser:

```
https://stream.mux.com/DS00Spx1CV902MCtPj5WknGlR102V5HFkDe.m3u8
```

if for some reason opening directly in the browser doesn't work, can you try:

```
<video controls src="https://stream.mux.com/DS00Spx1CV902MCtPj5WknGlR102V5HFkDe.m3u8"></video>
```

^ I'm curious if the same problem happens, this will help us debug

The way this SDK works right now is: If the browser supports Hls playback natively, then it uses the browser's native Hls implementation, otherwise it fallsback to Hls.js. In these cases Chrome and Edge for Android I believe support Hls natively so it should be using that. If that native Hls playback engine is particularly problematic we'll need to swap in Hls.js for those cases.

Thanks and we'll get this resolved asap

**Comment by agentile** [comment]
Hmm, we are seeing similar issues with Android, trying to narrow it down.

`VIDEOJS: ERROR: (CODE:4 MEDIA_ERR_SRC_NOT_SUPPORTED)`

https://stream.mux.com/DS00Spx1CV902MCtPj5WknGlR102V5HFkDe.m3u8

works on Samsung S20 ultra - Chrome version 92.0.4515.159

**Comment by agentile** [comment]
Reading other videojs threads

Is it worth doing any of these options

```
crossorigin="anonymous"
data-setup='{"html5": {"hls": {"overrideNative": true}}}'
```

Or is that not ideal with what the mux kit is trying to do?

We have something pretty barebones right now

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8" />
    <meta property="og:title" content="Mobile Upload Video" />
    <meta property="og:type" content="video.other" />
    <meta property="og:video" content="<?=$file ?>" />
    <meta property="og:video:secure_url" content="<?=$file ?>" />
    <meta property="og:image" content="<?php echo $thumbnail; ?>" />
    <meta property="og:image:type" content="image/png" />

    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Video Player</title>

    <script src="https://unpkg.com/@mux/videojs-kit@latest/dist/index.js"></script>

    <link rel="stylesheet" href="https://unpkg.com/@mux/videojs-kit@latest/dist/index.css">
</head>

<body class="minimal">
    <video
      id="my-player"
      class="video-js vjs-16-9"
      controls
      preload="auto"
      crossorigin="anonymous"
      width="100%"
      data-setup='{"html5": {"hls": {"overrideNative": true}}}'
    >
      <source src="<?=$playbackId ?>" type="video/mux" />
    </video>
</body>
</html>
```

**Comment by Chamersoun** [comment]
@dylanjha
We tried to open the link directly `https://stream.mux.com/DS00Spx1CV902MCtPj5WknGlR102V5HFkDe.m3u8`. The video works fine on all of the devices:

- Xiaomi Redmi Note 10S Firefox / Edge / Chrome
- Lenovo Tablet Firefox / Edge / Chrome
- Realme 7 Pro Firefox / Edge / Chrome
- iPhone XR - Firefox / Safari / Chrome

**Comment by dylanjha** [comment]
@agentile with regards to ` {"overrideNative": true}` -- I haven't verified this, but my best guess is that will be ignored by this SDK. That pertains to video-js's built-in HLS streaming engine (http-streaming), but this SDK swaps that out in favor of Hls.js.

@Chamersoun thanks -- that helps. So the native HLS playback engine is working just fine. Let me dive in a little more and see what's up. Something more nefarious is going on.

**Comment by dylanjha** [comment]
@agentile @Chamersoun fixed in [0.5.0](https://github.com/muxinc/videojs-mux-kit/releases/tag/v0.5.0)!
